Connectors/Rhino/Speckle.Connectors.GrasshopperShared/Operations/Send/GrasshopperSendOperation.cs

```diff
@@ -176,6 +185,47 @@ Collection currentColl
     }
   }
 
+  /// <summary>
+  /// Converts a <see cref="SpeckleDataObjectWrapper"/> to underlying DataObject with properly configured displayValue.
+  /// Processes colors and materials for each individual geometry during conversion.
+  /// </summary>
+  private DataObject UnwrapDataObject(
+    SpeckleDataObjectWrapper wrapper,
+    GrasshopperColorPacker colorPacker,
+    GrasshopperMaterialPacker materialPacker
+  )
+  {
+    DataObject dataObject = wrapper.DataObject;
+
+    // Convert geometries back to Base objects for displayValue
+    var displayValue = new List<Base>();
+    foreach (var geometryWrapper in wrapper.Geometries)
+    {
+      Base geometryBase = UnwrapGeometry(geometryWrapper);
+      displayValue.Add(geometryBase);
+
+      // process color and material for each geometry while we're iterating
+      // this could be in the switch statements (like SpeckleGeometryWrapper) but then we're unnecessarily looping twice
+      if (geometryWrapper.ApplicationId != null)
+      {
+        colorPacker.ProcessColor(geometryWrapper.ApplicationId, geometryWrapper.Color);
+        materialPacker.ProcessMaterial(geometryWrapper.ApplicationId, geometryWrapper.Material);
+      }
+    }
+
+    // Update the DataObject's displayValue
+    dataObject.displayValue = displayValue;
+
+    // Ensure properties are set from wrapper
+    Dictionary<string, object?> props = [];
+    if (wrapper.Properties.CastTo(ref props))
```

Comment:
hmm I don't think this is needed, the dataobject should already have its props set when the wrapper props are set.